ground_strike_feature.py

```diff
@@ -0,0 +1,375 @@
+
+        sum_a = self.get_ground_detail_sum(round_number, f'fighter_a_{column_name}' if not is_defense else f'fighter_b_{column_name}', fights_as_a, not is_percentage)
+        sum_b = self.get_ground_detail_sum(round_number, f'fighter_b_{column_name}' if not is_defense else f'fighter_a_{column_name}', fights_as_b, not is_percentage)
+
+        return (sum_a + sum_b), (len(fights_as_a) + len(fights_as_b))
+    
+    def get_ground_detail_sum(self, round_number, column_name, fights, per_minute=True):
+        """
+        Computes the sum from each of the fights for the ground strikes stats
+
+        Parameters:
+        - round_number (int): The round number of the current fight
+        - column_name (string): The name of the column to compute
+        - fights (pd.Dataframe): The dataframe containing all the past fights for the fighter
+        - per_minute (bool): Whether to compute the sum per minute or not
+
+        Returns:
+        - (float): The sum of the ground strikes stats
+        """
+        sum = 0
+
+        for _, row in fights.iterrows():
+            if row[column_name] == "":
+                continue
+            
+            if row["outcome_round"] is round_number:
+                divisor = self.convert_time_to_minutes(row["outcome_time"]) if per_minute else 1
+                sum += float(row[column_name]) / divisor
+            else:
+                divisor = 5 if per_minute else 1
+                sum += float(row[column_name]) / divisor
+        
+        return sum
+    
+    def convert_time_to_minutes(self, time):
+        """
+        Converts the time from the fights to minutes
+
+        Parameters:
+        - time (string): The time from the fights
+
+        Returns:
+        - (float): The time in minutes
+        """
+        minutes, seconds = map(int, time.split(':'))
+        total_minutes = minutes + seconds / 60
+        return total_minutes
+    
+    def create_col_names_significant_ground_strikes(self):
+        """
+        Generates column names for significant strike statistics.
+
+        Returns:
+            list: A list of strings, each representing a column name for a specific significant strike statistic.
+        """
+        col_names = []
+        fighters = ['fighter-a', 'fighter-b']
+        ground_strike_stats = ['ground-strikes-attempted-per-minute', 'ground-strikes-landed-per-minute', 'ground-strikes-accuracy-percentage', 'ground-strikes-absorbed-per-minute', 'ground-strikes-recieved-per-minute', 'ground-strikes-defended-percentage', 'ground-strikes-attempted-per-minute-diff', 'ground-strikes-landed-per-minute-diff', 'ground-strikes-accuracy-percentage-diff', 'ground-strikes-absorbed-per-minute-diff', 'ground-strikes-recieved-per-minute-diff', 'ground-strikes-defended-percentage-diff']
+        rounds = ['R1', 'R2', 'R3', 'R4', 'R5', 'overall']
+        time_periods = ['l3', 'l5', 'alltime']
+
```

Comment:
we should standardize the col names. I like the format you have here. we should have all the col names be named in the same way with the units at the end of the stat and l3, l5 instead of last-3-fights.  or whether to have diff or differential.

Reply:
I feel the shorter we can make it the better as long as it still makes sense